disambiguator.py

```diff
@@ -371,95 +385,143 @@ def group_categories():
+    post = np.zeros((0, num_col))
+
+    if s < 0:
+        pre = np.zeros((-s, num_col))
+        s = 0
 
+    if e > A.shape[0]:
+        post = np.zeros((e - A.shape[0], num_col))
+        e = A.shape[0]
 
-def init_random_params(scale, layer_sizes, rs=npr.RandomState(0)):
-    """
-    Build a list of (weights, biases) tuples,
-    one for each layer in the net.
+    if pad:
+        return np.concatenate((pre, A[s:e], post))
+    return A[s:e]
 
-    """
-    return [(scale * rs.randn(m, n),   # weight matrix
-             scale * rs.randn(n))      # bias vector
-            for m, n in zip(layer_sizes[:-1], layer_sizes[1:])]
 
+def make_grams(darrays, labels, num_grams, target_tag=EOS_PUNC, tag_order=None):
+    ''' Given darrays, only take the ones that have a certain
+        label. Then take k/2 neighbors from either side.
+
+        Args
+        ----
+        darrays : np array
+                 descriptor array
 
-def neural_net_predict(params, inputs):
-    """
-    Implements a deep neural network for classification.
-    params is a list of (weights, bias) tuples.
-    inputs is an (N x D) matrix.
-    returns normalized class log-probabilities.
+        labels : np array
+                 output array
 
-     """
-    for W, b in params:
-        outputs = np.dot(inputs, W) + b
-        inputs = np.tanh(outputs)
-    return outputs - logsumexp(outputs, axis=1, keepdims=True)
+        num_grams : integer
+                    num of surrounding grams
+                    if num_grams = 3, 3 on each side
+                    will be taken.
 
+        target_tag : string
+                     what tag to index to look for
 
-def l2_norm(params):
-    """
-    Computes l2 norm of params by flattening them into a vector.
+        tag_order : array
+                    list of tags
 
-    """
-    flattened, _ = flatten(params)
-    return np.dot(flattened, flattened)
+        Returns
+        -------
+        new_darrays : np array
+                     gram'd descriptor array
 
+        new_labels : np array
+                     gram'd labels
+    '''
 
-def log_posterior(params, inputs, targets, L2_reg):
-    log_prior = -L2_reg * l2_norm(params)
-    log_lik = np.sum(neural_net_predict(params, inputs) * targets)
-    return log_prior + log_lik
+    if tag_order is None:
+        tag_order = cPickle.load(
+            open('storage/brown_tag_order.pkl', 'rb'))
 
+    if not target_tag in tag_order:
+        raise ValueError('Target tag not found in tag order.')
 
-def accuracy(params, inputs, targets):
-    target_class = np.argmax(targets, axis=1)
-    predicted_class = np.argmax(neural_net_predict(params, inputs), axis=1)
-    return np.mean(predicted_class == target_class)
+    target_idx = np.where(tag_order == target_tag)[0][0]
+    eos_idx = np.where(darrays[:,target_idx] > 0)[0]
 
+    new_darrays = np.zeros((eos_idx.shape[0],darrays.shape[1]*(2*num_grams + 1)))
+    new_labels = np.zeros((eos_idx.shape[0], labels.shape[1]))
```

Comment:
I'm not sure what you're going for here. `labels` is a 1-D array when you call it from `create_features_labels`, so `.shape[1]` fails